### ISSUE TITLE
Disable premultiplied alpha when mipmapping illumination textures

### ISSUE THREAD

**Issue by osreboot** [bozbez/tm3-mod-exporter#3]
Just wanted to say that this is a great tool, and I used it extensively during the recent 20th anniversary contest.

Unfortunately, I noticed an issue with the mipmap generation for the self-illumination textures (BC3, ending in "_I"). In TM2020, the self-illumination alpha channel value of zero encodes to brake lights, however the color information in fully transparent regions doesn't seem to exist in lower-level mipmaps generated by this tool. Practically, this means that when exporting the default game illumination file for car details, the brake meter on the front wheels is missing unless the camera is really close to the car.

Example file with consistent RGB rows and horizontally varied alpha value:
<img src="https://github.com/bozbez/tm3-mod-exporter/assets/11065655/bf02afd1-d9fb-4d2c-9c60-c8effe51e13a.png" width="300">

First mipmap generated by GIMP BC3 .dds export (expected):
![image](https://github.com/bozbez/tm3-mod-exporter/assets/11065655/10a79f5f-d236-4b0a-8338-223dbd45bba7)

First mipmap generated by this tool:
![image](https://github.com/bozbez/tm3-mod-exporter/assets/11065655/4f147370-274f-42ea-ad3e-6cb16633059c)

I believe this is due to using premultiplied alpha for [all mipmap generation](https://github.com/bozbez/tm3-mod-exporter/blob/master/src/export_thread.cpp#L89), which can destroy RGB data for pixels with an alpha of zero. NVTT's sample code [recommends](https://github.com/nvpro-samples/nvtt_samples/blob/main/mipmap/main.cpp#L56-L58) avoiding premultiplication for textures where the alpha channel represents non-transparency material data. Because you're already using file suffixes to automatically determine the compression type, it seems like you could also use this to disable premultiplied alpha when mipmapping illumination textures.

I need C++ practice and I'd love to implement this change myself if you think it's appropriate!

**Comment by bozbez** [comment]
Yeah go for it, I'd make a small helper class that wraps nvtt::Format and a premultiply alpha bool and move GuessFormat and FormatToString into it.
